src/passes/arithmetic/Arithmetic.cpp

```diff
@@ -261,13 +261,13 @@ PreservedAnalyses Arithmetic::run(Module &M,
     opts_.insert({F, std::move(opt)});
 
     for (BasicBlock& BB : *F) {
-      Changed |= runOnBasicBlock(BB);
+      bool Changed = runOnBasicBlock(BB);
+      ModuleChanges.notify(Changed);
```

Comment:
Why per basic block? I'd expect to have `ModuleChanges.notify(Changed);` before `return ModuleChanges.report();`, and here still have `Changed |= ...`.

Reply:
Well, we'd need yet another variable to track per-BB changes. `ModuleChanges` now tracks it and I removed `Changed`.